mvc/src/main/java/nextstep/mvc/controller/tobe/HandlerExecution.java

```diff
@@ -2,11 +2,31 @@
 
 import jakarta.servlet.http.HttpServletRequest;
 import jakarta.servlet.http.HttpServletResponse;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Arrays;
 import nextstep.mvc.view.ModelAndView;
 
 public class HandlerExecution {
 
+    private final Method method;
+
+    public HandlerExecution(final Method method) {
+        this.method = method;
+    }
+
     public ModelAndView handle(final HttpServletRequest request, final HttpServletResponse response) throws Exception {
-        return null;
+        final Object controller = findController();
+        return (ModelAndView) method.invoke(controller, request, response);
+    }
+
+    private Object findController()
+            throws InstantiationException, IllegalAccessException, InvocationTargetException, NoSuchMethodException {
+        return Arrays.stream(method.getDeclaringClass()
+                        .getConstructors())
+                .filter(it -> it.getParameterCount() == 0)
+                .findFirst()
+                .orElseThrow(() -> new NoSuchMethodException("기본 생성자가 존재하지 않습니다."))
+                .newInstance();
```

Comment:
c: `getConstructor` 메서드의 설명을 보면 이렇게 나와있습니다.
```
Returns: the Constructor object of the public constructor that matches the specified parameterTypes
```
따라서 굳이 모든 생성자를 가져와 filter로 거를 필요 없이 아래와 같이 호출하면 파라미터를 받지 않는 빈 생성자를 가져올 수 있습니다.

```suggestion
        return method.getDeclaringClass()
                .getConstructor()
                .newInstance();
```

Reply:
올ㅋ